### PR TITLE
[UnbreakingFetch][1/n]: Allow adding / toggling SignalExchangeAPIs

### PR DESCRIPTION
**Pull request by schatten**
Summary
---
Hashing has been made to work with pytx-1.0. Fetchers are next, because without them, matchers can't function.

For fetchers to work, they need to get all collabs. To get all collabs, we need to get all supported signal exchanges.

This PR adds toggle-able signal exchanges.

You can add a signal-exchange using their fully-qualified class name. By default, add FBThreatExchange and LocalFile.

Test Plan
---
Added unit tests.

Ran migration using `$ terraform -chdir=terraform apply`.

Then, in console, verified that both are 'enabled'.

```
$ python
>>> [
   t.to_concrete_class()
   for t in ToggleableSignalExchangeAPIConfig.get_all()
   if t.enabled
]
[
  <class 'threatexchange.exchanges.impl.fb_threatexchange_api.FBThreatExchangeSignalExchangeAPI'>,
  <class 'threatexchange.exchanges.impl.file_api.LocalFileSignalExchangeAPI'>
]
```

### CODE REVIEW

File: hasher-matcher-actioner/hmalib/common/configs/tx_apis.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    signal_exchange_api_class: str
+
+    # Allow soft disables
+    enabled: bool = True
+
+    @staticmethod
+    def get_name_from_type(signal_exchange_api: t.Type[SignalExchangeAPI]) -> str:
+        return f"signal_exchange_api:{full_class_name(signal_exchange_api)}"
+
+    def to_concrete_class(self):
```
Comment (Dcallies) [inline hasher-matcher-actioner/hmalib/common/configs/tx_apis.py:33]: (nit): base class could be the return type here

File: hasher-matcher-actioner/hmalib/common/configs/tx_apis.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+@dataclass
+class ToggleableSignalExchangeAPIConfig(HMAConfig):
+    signal_exchange_api_class: str
+
+    # Allow soft disables
+    enabled: bool = True
+
+    @staticmethod
+    def get_name_from_type(signal_exchange_api: t.Type[SignalExchangeAPI]) -> str:
+        return f"signal_exchange_api:{full_class_name(signal_exchange_api)}"
```
Comment (Dcallies) [inline hasher-matcher-actioner/hmalib/common/configs/tx_apis.py:31]: (ignorable): I went through all that trouble to guarantee you human readable names, and you throw it away! :P

**Review by Dcallies — APPROVED** [review]
I have a non-blocking TODO to do the same thing on the CLI (the reason why `tx config signal` and `tx config content` exists).

File: hasher-matcher-actioner/hmalib/common/configs/tx_apis.py (hunk 2 above)
Comment (schatten) [reply hasher-matcher-actioner/hmalib/common/configs/tx_apis.py:31]: This is only for config storage. Not shown to humans.

**Pull request merged**